hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/CryptoGetAccountRecordsHandler.java

```diff
@@ -37,9 +46,12 @@
  */
 @Singleton
 public class CryptoGetAccountRecordsHandler extends PaidQueryHandler {
+    private final RecordCache recordCache;
+
     @Inject
-    public CryptoGetAccountRecordsHandler() {
+    public CryptoGetAccountRecordsHandler(@NonNull final RecordCache recordCache) {
         // Exists for injection
+        this.recordCache = requireNonNull(recordCache);
```

Comment:
I think we have to get the record cache from the query context.  Otherwise how do we know we have the correct record cache?  I don't _think_ it's a global singleton in the modularized world (but perhaps check with Michael H. or Richard to be certain).

Reply:
Checked with @rbair23. Looks like the record cache is a singleton